tests/run-make/x86_64-fortanix-unknown-sgx-lvi/rmake.rs

```diff
@@ -0,0 +1,96 @@
+// ignore-tidy-linelength
+// Reason: intel.com link
+
+// This security test checks that the disassembled form of certain symbols
+// is "hardened" - that means, the assembly instructions match a pattern that shows
+// lack of vulnerability to a Load Value Injection attack.
+// To do so, a test crate is compiled, and certain symbols are found, disassembled
+// and checked one by one.
+// See https://github.com/rust-lang/rust/pull/77008
+
+// On load value injection:
+// https://www.intel.com/content/www/us/en/developer/articles/technical/software-security-guidance/technical-documentation/load-value-injection.html
+
+//@ only-x86_64-fortanix-unknown-sgx
+
+use run_make_support::{cmd, cwd, llvm_filecheck, llvm_objdump, regex, set_current_dir, target};
+
+fn main() {
+    let main_dir = cwd();
+    set_current_dir("enclave");
+    // HACK(eddyb) sets `RUSTC_BOOTSTRAP=1` so Cargo can accept nightly features.
+    // These come from the top-level Rust workspace, that this crate is not a
+    // member of, but Cargo tries to load the workspace `Cargo.toml` anyway.
+    cmd("cargo")
+        .env("RUSTC_BOOTSTRAP", "1")
+        .arg("-v")
+        .arg("run")
+        .arg("--target")
+        .arg(target())
+        .run();
+    set_current_dir(&main_dir);
+    check("unw_getcontext", "unw_getcontext.checks");
+    check("__libunwind_Registers_x86_64_jumpto", "jumpto.checks");
+
+    check("std::io::stdio::_print::[[:alnum:]]+", "print.with_frame_pointers.checks");
+
+    check("st_plus_one_global_asm", "rust_plus_one_global_asm.checks");
+
+    check("_plus_one_c", "cc_plus_one_c.checks");
+    check("_plus_one_c_asm", "cc_plus_one_c_asm.checks");
+    check("_plus_one_cxx", "cc_plus_one_cxx.checks");
+    check("_plus_one_cxx_asm", "cc_plus_one_cxx_asm.checks");
+    check("_plus_one_asm", "cc_plus_one_asm.checks");
+
+    check("cmake_plus_one_c", "cmake_plus_one_c.checks");
+    check("cmake_plus_one_c_asm", "cmake_plus_one_c_asm.checks");
+    check("cmake_plus_one_c_global_asm", "cmake_plus_one_c_global_asm.checks");
+    check("cmake_plus_one_cxx", "cmake_plus_one_cxx.checks");
+    check("cmake_plus_one_cxx_asm", "cmake_plus_one_cxx_asm.checks");
+    check("cmake_plus_one_cxx_global_asm", "cmake_plus_one_cxx_global_asm.checks");
+    check("cmake_plus_one_asm", "cmake_plus_one_asm.checks");
+}
+
+fn check(func_re: &str, checks: &str) {
+    let dump = llvm_objdump()
+        .input("enclave/target/x86_64-fortanix-unknown-sgx/debug/enclave")
+        .args(&["--syms", "--demangle"])
+        .run()
+        .stdout_utf8();
+    let re = regex::Regex::new(&format!("[[:blank:]]+{func_re}")).unwrap();
+    let func = re.find_iter(&dump).map(|m| m.as_str().trim()).collect::<Vec<&str>>().join(",");
+    let dump = llvm_objdump()
+        .input("enclave/target/x86_64-fortanix-unknown-sgx/debug/enclave")
+        .args(&["--syms", &format!("--disassemble-symbols={func}")])
+        .run()
+        .stdout_utf8();
```

Comment:
The script does this:
```
    func="$(${objdump} --syms --demangle "${enclave}" | \
            grep --only-matching -E "[[:blank:]]+${func_re}\$" | \
            sed -e 's/^[[:space:]]*//' )"
```
This is needed. Otherwise you get output like:
```                                                                                                                                                                                                                                            
enclave/target/x86_64-fortanix-unknown-sgx/debug/enclave:       file format elf64-x86-64                                                                                                                                                    
                                                                                                                                                                                                                                            
SYMBOL TABLE:                                                                                                                                                                                                                               
0000000000000000 l    df *ABS*  0000000000000000 30q2fp73pc99bmzfe1chpqhcp                                                                                                                                                                  
0000000000012600 l     F .text  000000000000001e .hidden _ZN3std3sys9backtrace28__rust_begin_short_backtrace17hb9bbcbf22e1d0f53E                                                                                                            
0000000000012620 l     F .text  0000000000000006 .hidden __llvm_lvi_thunk_r11 
...
000000000000000c l *ABS* 0000000000000000 tcsls_user_mxcsr                                                                                                                                                                     
0000000000000010 l *ABS* 0000000000000000 tcsls_last_rsp                                                                                                                                                                       
0000000000000018 l *ABS* 0000000000000000 tcsls_panic_last_rsp                                                                                                                                                                 
0000000000000020 l *ABS* 0000000000000000 tcsls_debug_panic_buf_ptr                                                                                                                                                            
0000000000000028 l *ABS* 0000000000000000 tcsls_user_rsp                                                                                                                                                                       
0000000000000030 l *ABS* 0000000000000000 tcsls_user_retip    # <--- false positive on this ret
...
Disassembly of section .text:                                                                                                                                                                                                               
                                                                                                                                                                                                                                            
0000000000031a28 <__libunwind_Registers_x86_64_jumpto>:                                                                                                                                                                                     
   31a28: 48 8b 47 38                   movq    0x38(%rdi), %rax                                                                                                                                                                            
   31a2c: 0f ae e8                      lfence                                                                                                                                                                                              
   31a2f: 48 83 e8 10                   subq    $0x10, %rax                                                                                                                                                                                 
   31a33: 48 89 47 38                   movq    %rax, 0x38(%rdi)                                                                                                                                                                            
   31a37: 48 8b 5f 20                   movq    0x20(%rdi), %rbx                                                                                                                                                                            
   31a3b: 0f ae e8                      lfence   
```
and that gives false positives when checking that `ret` is not present

Reply:
In the original script:

```sh
    func="$(${objdump} --syms --demangle "${enclave}" | \
            grep --only-matching -E "[[:blank:]]+${func_re}\$" | \
            sed -e 's/^[[:space:]]*//' )"
    ${objdump} --disassemble-symbols="${func}" --demangle \
      "${enclave}" > "${asm}"
```

The first objdump seeks all `func_re` instances with a leading space, then trims away the space. The second objdump has no further filtering or normalization.

In the Rust code:

```rs
    let dump = llvm_objdump()
        .input("enclave/target/x86_64-fortanix-unknown-sgx/debug/enclave")
        .args(&["--syms", "--demangle"])
        .run()
        .stdout_utf8();
    let re = regex::Regex::new(&format!("[[:blank:]]+{func_re}")).unwrap();
    let func = re.find_iter(&dump).map(|m| m.as_str().trim()).collect::<Vec<&str>>().join(",");
    let dump = llvm_objdump()
        .input("enclave/target/x86_64-fortanix-unknown-sgx/debug/enclave")
        .args(&["--syms", &format!("--disassemble-symbols={func}")])
        .run()
        .stdout_utf8();
```

The regex pattern is `[[:blank:]]+{func_re}`, which should avoid detecting the `tcsls_user_retip`, and then the `.trim()` replaces the `sed` of the original script. As for the second objdump, it has no filtering or normalizations like the original script.

Let me know if I missed something.